sqlalchemy_bigquery/base.py

```diff
@@ -655,25 +656,79 @@ def get_column_specification(self, column, **kwargs):
+
+            assert isinstance(
+                partition_expr, str
+            ), f"bigquery_partitioning dialect option accepts only {str}, provided {repr(partition_expr)}"
+
+            partition_by_or_cluster_by_or_options.append(
+                f"PARTITION BY {partition_expr}"
             )
-            opts.append(f"description={description}")
 
-        if "friendly_name" in bq_opts:
-            opts.append(
-                "friendly_name={}".format(
-                    process_string_literal(bq_opts["friendly_name"])
-                )
+        if "clustering_fields" in bq_opts:
+            clustering_fields = bq_opts.get("clustering_fields")
+
+            assert isinstance(
+                clustering_fields, list
+            ), f"bigquery_clustering_fields dialect option accepts only {list}, provided {repr(clustering_fields)}"
+
+            for field in clustering_fields:
+                if field not in table.c:
+                    raise NoSuchColumnError(field)
+
+            partition_by_or_cluster_by_or_options.append(
+                f"CLUSTER BY {', '.join(clustering_fields)}"
             )
 
-        if opts:
-            return "\nOPTIONS({})".format(", ".join(opts))
+        options = {}
 
-        return ""
+        if ("description" in bq_opts) or table.comment:
+            description = bq_opts.get("description", table.comment)
+            options["description"] = description
+
+        table_option_list = {
+            "friendly_name": str,
+            "expiration_timestamp": datetime.datetime,
+            "partition_expiration_days": int,
+            "require_partition_filter": bool,
+            "default_rounding_mode": str,
+        }
+
+        for table_option in table_option_list:
+            if table_option in bq_opts:
+                options[table_option] = bq_opts.get(table_option)
+
+        if options:
+
+            def process_option_value(option, value):
+                if option in table_option_list:
+                    assert isinstance(
+                        value, table_option_list[option]
+                    ), f"bigquery_{option} dialect option accepts only {table_option_list[option]}, provided {repr(value)}"
+                else:
+                    return process_string_literal(value)
+
+                if table_option_list[option] == str:
+                    return process_string_literal(value)
+                if table_option_list[option] == int:
+                    return int(value)
+                if table_option_list[option] == bool:
+                    return "true" if value else "false"
+                if table_option_list[option] == datetime.datetime:
+                    return BQTimestamp.process_timestamp_literal(value)
+
+            options_list = [
+                f"{k}={process_option_value(k,v)}" for (k, v) in options.items()
+            ]
+            options_str = f"OPTIONS({', '.join(options_list)})"
```

Comment:
Please rename this to represent what it is a bit more clearly:

```suggestion
            complete_option_statement = f"OPTIONS({', '.join(options_list)})"
```

Reply:
I will finally drop this intermediate function 

```py
            individual_option_statements = [
                f"{k}={process_option_value(k,v)}" for (k, v) in options.items()
            ]

            clauses.append(f"OPTIONS({', '.join(individual_option_statements)})")
```

As black formats the code in a way that is not ideal when using a long var name (with extra parentheses):

```py
            individual_option_statements = [
                f"{k}={process_option_value(k,v)}" for (k, v) in options.items()
            ]
            complete_option_statement = (
                f"OPTIONS({', '.join(individual_option_statements)})"
            )
            clauses.append(complete_option_statement)
```

Reply:
Hum, nerver mind, we are not using `black` but `flake8` here.